frontend/docsettings.lua

```diff
@@ -592,4 +609,65 @@ function DocSettings:flushCustomMetadata(doc_path)
     end
 end
 
+-- hash-based SDR storage
+
+local function getSdrsInDir(path)
+    -- Get all the metadata.filetype.lua files under directory path.
+    -- Derived from readerdictionary.getIfosInDir()
+    local sdrs = {}
+    local ok, iter, dir_obj = pcall(lfs.dir, path)
+    if ok then
+        for name in iter, dir_obj do
+            if name ~= "." and name ~= ".." then
+                local fullpath = path.."/"..name
+                local attributes = lfs.attributes(fullpath)
+                if attributes ~= nil then
+                    if attributes.mode == "directory" then
+                        local dirifos = getSdrsInDir(fullpath) -- recurse
+                        for _, ifo in pairs(dirifos) do
+                            table.insert(sdrs, ifo)
+                        end
+                    elseif fullpath:match("metadata%..+%.lua$") then
+                        table.insert(sdrs, fullpath)
+                    end
+                end
+            end
+        end
+    end
+    return sdrs
+end
+
+
+function DocSettings.getHashDirSdrInfos()
+   local sdrs = getSdrsInDir(DOCSETTINGS_HASH_DIR)
+   local title_author_strs = {}
+   for _, sdr in ipairs(sdrs) do
+    -- Ignore empty files
+    if lfs.attributes(sdr, "size") > 0 then
+        local ok, stored
+        ok, stored = pcall(dofile, sdr)
+        -- Ignore empty tables
+        if ok and next(stored) ~= nil then
+            local m = DocSettings:extend{}
+            m.data = stored
```

Comment:
It looks like you don't use `m` - and keep just using `stored`, which is fine.
So these 2 lines seems uneeded.